llvm/lib/SYCLLowerIR/PrepareSYCLNativeCPU.cpp

```diff
@@ -306,10 +277,36 @@ static Function *addReplaceFunc(Module &M, StringRef Name, Type *StateType) {
       Builder.CreateGEP(StateType, F->getArg(1), {Zero, Offset, IdxProm});
   auto *Load = Builder.CreateLoad(I64Ty, GEP);
   Builder.CreateRet(Load);
-  F->setLinkage(GlobalValue::LinkageTypes::WeakAnyLinkage);
```

Comment:
Do I understand correctly that we now emit those functions as `external` ones? Should we maybe make them `internal` instead to enable more aggressive optimizations with them? If those functions are external, they may as well be defined in a library written in C++ instead of being on-demand generated at LLVM IR level

Reply:
You are right, setting `internal` makes most sense here, I've done it

Reply:
Note that no changes were done to this file, I still see the same removed line without any replacement - a missed `git commit` perhaps?

Reply:
I've made the change at the end of `addReplaceFunc` [here](https://github.com/intel/llvm/blob/faa3be6ba1f969c583091d5bb2d7c5267ac4fb74/llvm/lib/SYCLLowerIR/PrepareSYCLNativeCPU.cpp#L307) so that it applies to all the functions that we insert in the module